### PR TITLE
Fix hint value assignment in SetupUnitConversion() and check if there are any diffs

### LINKED ISSUE #10260: Hint value assignment typo in SetupUnitConversion()
Issue overview
--------------
In `SetupUnitConversion()`, there are two consecutive lines that assign the same variable: 

https://github.com/NREL/EnergyPlus/blob/7636e6b3e972dad33c41815304938b2139dde367/src/EnergyPlus/OutputReportTabular.cc#L18448-L18449

image

By inferring from the context, the second line should be assigned to `ort->UnitConv(94).hint` instead of `ort->UnitConv(93).hint`; otherwise the first line would not be needed at all.

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version): All
 - Version of EnergyPlus (if using an intermediate build, include SHA): v23.2, or may in earlier versions as well
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jcyuan2020**
Pull request overview
---------------------
 - Fixes #10260
 - Trying to check if it would cause any diffs
 - Actually, it does cause a noticeable diff in one of the LEED tables in 12 test files:

Diff Summary:
image

Before:
image

After:
image

Since the `.hint` value is used to look up the correct units to go when there are multiples, this should be a confirmed bug that has an impact and it needs to be fixed.

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/OutputReportTabular.cc [hunk 1]
```diff
@@ -18446,7 +18446,7 @@ void SetupUnitConversions(EnergyPlusData &state)
     ort->UnitConv(91).hint = "COOL";
     ort->UnitConv(92).hint = "ELEC";
     ort->UnitConv(93).hint = "GAS";
-    ort->UnitConv(93).hint = "ADDITIONAL";
+    ort->UnitConv(94).hint = "ADDITIONAL";
```
Comment (mjwitte) [inline src/EnergyPlus/OutputReportTabular.cc:18449]: Clearly the correct fix.

File: tst/EnergyPlus/unit/OutputReportTabular.unit.cc [hunk 2]
```diff
@@ -13043,3 +13043,48 @@ TEST_F(SQLiteFixture, OutputReportTabular_DistrictHeating)
         EXPECT_NEAR(DistrictHeatingSteam * 2 / 3.6e6, return_val, 0.01) << "Failed for query: " << query;
     }
 }
+
+TEST_F(EnergyPlusFixture, OutputReportTabular_Test_SetupUnitConversion_Fix)
```
Comment (mjwitte) [inline tst/EnergyPlus/unit/OutputReportTabular.unit.cc:13047]: Confirmed that the unit test fails with develop.

**Review by mjwitte — APPROVED** [review]
@jcyuan2020 Ran 5ZoneAutoDXVAV.idf before and after, confirmed changes in table heading and values.
This looks good to go.
@Myoldmopar Tagx2!

**Comment by jcyuan2020** [comment]
@mjwitte @Myoldmopar It is suggested to get this pr merged before PR #10256.

**Comment by Myoldmopar** [comment]
Got it @jcyuan2020.  I just merged a big PR, but it didn't appear to have any Git conflicts.  I'll still build this branch with develop as of right now and make sure everything builds/tests happily before continuing.

**Comment by Myoldmopar** [comment]
This still passes happily locally.  Let's get this one in now.  Thanks @jcyuan2020

**Pull request merged**